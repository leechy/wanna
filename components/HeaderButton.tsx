import { useThemeColor } from '@/hooks/useThemeColor';

import { Text, TouchableOpacity, StyleSheet, Platform, StyleProp, ViewStyle } from 'react-native';
import { SvgProps } from 'react-native-svg';

interface HeaderButtonProps {
  title: string;
  icon?: React.FC<SvgProps>;
  iconPosition?: 'start' | 'end';
  onPress: () => void;
  color?: string;
  style?: StyleProp<ViewStyle>;
  accessibilityLabel?: string;
  accessibilityHint?: string;
}

export default function HeaderButton({
  title,
  icon: Icon,
  iconPosition = 'start',
  onPress,
  color,
  style,
  accessibilityLabel = 'Go back',
  accessibilityHint = 'Press to go to the previous screen',
}: HeaderButtonProps) {
  const primaryColor = useThemeColor({}, 'primary');

  return (
    <TouchableOpacity
      style={[styles.container, style || {}]}
      onPress={onPress}
      accessibilityRole="button"
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
      hitSlop={{ top: 15, bottom: 15, left: 15, right: 15 }}
    >
      {Icon && iconPosition === 'start' && <Icon width={24} height={24} color={color || primaryColor} />}
      <Text
        style={[
          styles.label,
          {
            color: color || primaryColor,
            fontSize: Platform.OS === 'ios' ? 16 : 18,
          },
        ]}
      >
        {title}
      </Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    margin: 0,
    padding: 0,
    flexDirection: 'row',
    alignItems: 'center',
  },
  label: {
    fontFamily: 'Montserrat',
    lineHeight: 24,
    fontWeight: '500',
  },
});
