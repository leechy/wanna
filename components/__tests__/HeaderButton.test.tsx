import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import HeaderButton from '../HeaderButton';

describe('HeaderButton', () => {
  it('renders correctly', () => {
    const { getByText } = render(
      <HeaderButton title="Click me" onPress={() => {}} />
    );
    expect(getByText('Click me')).toBeTruthy();
  });

  it('calls onPress when pressed', () => {
    const onPressMock = jest.fn();
    const { getByText } = render(
      <HeaderButton title="Click me" onPress={onPressMock} />
    );
    const button = getByText('Click me');

    fireEvent.press(button);
    expect(onPressMock).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <HeaderButton title="Click me" onPress={() => {}} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
