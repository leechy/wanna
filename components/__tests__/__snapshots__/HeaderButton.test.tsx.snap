// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "margin": 0,
      "opacity": 1,
      "padding": 0,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Montserrat",
          "fontWeight": "500",
          "lineHeight": 24,
        },
        {
          "color": "#3a1fc1",
          "fontSize": 16,
        },
      ]
    }
  >
    Click me
  </Text>
</View>
`;
